File: youlexuan-manager-web/src/main/java/com/offcn/sellergoods/controller/SpecificationController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.entity.SpecificationVo;
import com.offcn.pojo.TbSpecification;
import com.offcn.sellergoods.service.SpecificationService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/*
规格控制层
 */
@RestController
@RequestMapping("/specification")
public class SpecificationController {
    @Reference
    private SpecificationService specificationService;


    /*
    全部规格列表
     */
    @RequestMapping("/selectAll")
    public List<TbSpecification> selctAll() {
        return specificationService.selectAll();
    }

    /*
    分页列表
     */
    @RequestMapping("/selectPage")
    public PageResult selectPage(int pageNum, int pageSize) {
        return specificationService.selectPage(pageNum, pageSize);
    }

    /*
     增加规格
    */
    @RequestMapping("/insert")
    public Result insert(@RequestBody SpecificationVo specificationVo) {
        try {
            specificationService.insert(specificationVo);
            return new Result(true, "增加成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "增加失败");
        }
    }

    /*
    修改规格
     */
    @RequestMapping("/update")
    public Result update(@RequestBody SpecificationVo specificationVo) {
        try {
            specificationService.update(specificationVo);
            return new Result(true, "修改成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "修改失败");
        }
    }

    /*
    根据ID获取实体
     */
    @RequestMapping("/selectOne")
    public SpecificationVo selectOne(Long id) {
        return specificationService.selectOne(id);
    }

    /*
    删除规格
     */
    @RequestMapping("/delete")
    public Result delete(Long[] ids) {
        try {
            specificationService.delete(ids);
            return new Result(true, "删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(true, "删除失败");
        }
    }

    /*
    条件搜索
     */
    @RequestMapping("/search")
    public PageResult search(int page, int rows, @RequestBody TbSpecification tbSpecification) {
        return specificationService.findPage(tbSpecification, page, rows);
    }
}

File: youlexuan-manager-web/src/main/webapp/js/controller/specificationController.js
/*
定义controller
 */
app.controller('specificationController', function ($scope, $http, specificationService,$controller) {

    //声明:specificationController需要继承baseControler中的组件
    $controller('baseController', {$scope: $scope});

    //获取所有的规格信息
    $scope.selectAll = function () {
        specificationService.selectAll().success(function (response) {
            $scope.specificationList = response;
        })
    };

    //分页
    $scope.selectPage = function (page, rows) {
        specificationService.selectPage(page, rows).success(function (response) {
            $scope.specificationList = response.rows;
            $scope.paginationConf.totalItems = response.total;
        })
    };

    //增加/更新规格
    $scope.save = function () {
        if ($scope.entity.specification.id != null) {
            specificationService.update($scope.entity).success(function (response) {
                if (response.success) {
                    $scope.reloadList();
                    $scope.entity = {specification: {}, specificationOption:[]};
                }
            })
        } else {
            specificationService.insert($scope.entity).success(function (response) {
                if (response.success) {
                    $scope.reloadList();
                    $scope.entity = {specification: {}, specificationOption: []};
                }
            })
        }
    };

    //查询单个
    $scope.selectOne = function (id) {
        specificationService.selectOne(id).success(
            function (response) {
                $scope.entity = response;
            }
        )
    };

    //删除规格
    $scope.delete = function () {
        if ($scope.selectIds == null || $scope.selectIds.length == 0) {
            alert("请选择后进行删除");
            return;
        }
        specificationService.delete($scope.selectIds).success(
            function (response) {
                if (response.success) {
                    $scope.reloadList();
                    $scope.selectIds = [];
                } else {
                    alert(response.message);
                }
            }
        )
    };

    //条件搜索
    $scope.searchEntity = {};
    $scope.search = function (page, rows) {
        specificationService.search(page, rows, $scope.searchEntity).success(
            function (response) {
                $scope.paginationConf.totalItems = response.total;
                $scope.specificationList = response.rows;
            }
        )
    };

    $scope.entity = {specification: {}, specificationOption: []};
    //增加行
    $scope.addRow = function () {
        $scope.entity.specificationOption.push({});
    };

    //删除行
    $scope.deleteRow = function (index) {
        $scope.entity.specificationOption.splice(index, 1);
    };

})